src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IDictionaryConverter.cs

```diff
@@ -68,31 +65,41 @@ protected internal override bool OnWriteResume(Utf8JsonWriter writer, TCollectio
             }
             else
             {
-                Debug.Assert(state.Current.CollectionEnumerator is IDictionaryEnumerator);
                 enumerator = (IDictionaryEnumerator)state.Current.CollectionEnumerator;
             }
 
             JsonConverter<object?> converter = GetValueConverter(ref state);
             do
             {
-                if (enumerator.Key is string key)
+                if (ShouldFlush(writer, ref state))
```

Comment:
So this was the only collection we didn't apply the "ShouldFlush" pattern to?

Reply:
Yup.